src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -50,28 +60,177 @@
+   */
+  public List<String> rangeToDateList(final String range) {
+    LocalDate endDate = LocalDate.now();
+    LocalDate startDate;
+
+    if (range.equalsIgnoreCase("ytd")) {
+      startDate = LocalDate.ofYearDay(endDate.getYear(), 1);
+    } else {
+      // Confirms range is valid entry
+      String pattern = "\\d+[dmyDMY]";
+      Pattern p = Pattern.compile(pattern);
+      Matcher m = p.matcher(range);
+      if (!m.matches()) {
+        throw new IllegalArgumentException("Improper Range Provided.");
+      }
+      // Parses string to correct start date
+      String tp = range.substring(range.length() - 1);
+      int numOfTime = Integer.parseInt(range.substring(0,range.length()-1));
+      if (tp.equalsIgnoreCase("d")) {
+        startDate = endDate.minusDays(numOfTime);
+      } else if (tp.equalsIgnoreCase("m")) {
+        startDate = endDate.minusMonths(numOfTime);
+      } else {
+        startDate = endDate.minusYears(numOfTime);
+      }
+    }
+
+    // Predicate for checking on weekends
+    Predicate<LocalDate> isWeekend = date -> date.getDayOfWeek() == DayOfWeek.SATURDAY
+        || date.getDayOfWeek() == DayOfWeek.SUNDAY;
+
+    // List of Business Days
+    List<LocalDate> businessDays = startDate.datesUntil(endDate)
+        .filter(isWeekend.negate())
+        .collect(Collectors.toList());
+
+    // Convert to list of strings in proper format
+    List<String> dateList = new ArrayList<>();
+    for (LocalDate lD : businessDays) {
+      dateList.add(lD.format(DateTimeFormatter.ofPattern("yyyyMMdd")));
+    }
+    return dateList;
+  }
+
+  /**
+   * Chooses whether to get the historical price for the symbol from the repository or Iex.
+   * Precedence is always given to the repository, unless is not found within it.
+   *
+   * @param symbol Stock symbol to get the historical price for.
+   * @param date Input date (formatted YYYYMMDD) to get price data for.
+   *      Note: We will convert this into YYYY-MM-DD format for repository search.
+   *      - Format of Iex API output
+   * @return a list (length = 1) of IexHistoricalPrices objects.
+   */
+  public List<IexHistoricalPrices> getHistoricalPriceBySymbolAndDate(
+      final String symbol, final String date) {
+    //First, try the database
+    List<IexHistoricalPrices> dbList = new ArrayList<>();
+    List<IexHistoricalPricesDB> entry =
+        repository.findBySymbolAndDate(symbol, convertDateFormatToOutput(date));
+
+    if (!entry.isEmpty()) {
+      dbList.add(new IexHistoricalPrices(entry.get(0)));
+      return dbList;
+    } else {
+      //If not in database, call from Iex, insert into database, and return
+      List<IexHistoricalPrices> call = iexClient.getHistoricalPricesForSymbolByDate(symbol, date);
+      if (!call.isEmpty()) {
+        repository.save(new IexHistoricalPricesDB(call.get(0)));
+      }
+      return call;
+    }
+  }
+
+  /**
+   * Converts input date (format:YYYYMMDD) to output date (format:YYYY-MM-DD).
```

Comment:
These are standard formats. You can use java.util.Date and related classes (I believe the `DateFormatter` or something similar?). The YYYY-MM-DD format is known as the ISO standard format.

Reply:
Thank you - I didn't know about all of that. I updated this function to do this more smoothly.